contracts/Overtime/SportsAMMV2.sol

```diff
@@ -445,61 +508,91 @@ contract SportsAMMV2 is Initializable, ProxyOwned, ProxyPausable, ProxyReentranc
         if (exactReceived > _buyInAmount) {
-            defaultCollateral.safeTransfer(safeBox, exactReceived - _buyInAmount);
+            if (lqPool != address(0)) {
+                // Note: if needed add the logic to convert the surplus to default collateral and send to safeBox
+                IERC20(_collateral).safeTransfer(safeBox, exactReceived - _buyInAmount);
+            } else {
+                defaultCollateral.safeTransfer(safeBox, exactReceived - _buyInAmount);
+            }
         }
     }
 
     function _trade(ISportsAMMV2.TradeData[] memory _tradeData, TradeDataInternal memory _tradeDataInternal) internal {
         uint totalQuote = (ONE * _tradeDataInternal._buyInAmount) / _tradeDataInternal._expectedPayout;
         uint payout = _tradeDataInternal._expectedPayout;
-        //TODO: include this in the tradeQuote method for live trading
         uint fees = (safeBoxFee * _tradeDataInternal._buyInAmount) / ONE;
-        uint[] memory amountsToBuy = new uint[](_tradeData.length);
 
         if (!_tradeDataInternal._isLive) {
-            (totalQuote, payout, fees, amountsToBuy, ) = _tradeQuote(_tradeData, _tradeDataInternal._buyInAmount, false);
-        } else {
-            amountsToBuy[0] = payout;
+            (totalQuote, payout, fees, , ) = _tradeQuote(_tradeData, _tradeDataInternal._buyInAmount, false, 0);
         }
 
         uint payoutWithFees = payout + fees;
-
-        riskManager.checkAndUpdateRisks(_tradeData, _tradeDataInternal._buyInAmount);
-        riskManager.checkLimits(
-            _tradeDataInternal._buyInAmount,
-            totalQuote,
-            payout,
-            _tradeDataInternal._expectedPayout,
-            _tradeDataInternal._additionalSlippage
-        );
-
-        if (_tradeDataInternal._sendDefaultCollateral) {
-            defaultCollateral.safeTransferFrom(
-                _tradeDataInternal._requester,
-                address(this),
-                _tradeDataInternal._buyInAmount
+        uint transformDecimal = (18 - defaultCollateralDecimals);
+        if (_tradeDataInternal._collateralPool == address(0)) {
+            riskManager.checkAndUpdateRisks(_tradeData, _tradeDataInternal._buyInAmount);
+            riskManager.checkLimits(
+                _tradeDataInternal._buyInAmount,
+                totalQuote,
+                payout,
+                _tradeDataInternal._expectedPayout,
+                _tradeDataInternal._additionalSlippage
+            );
+            if (_tradeDataInternal._collateral == address(0)) {
+                defaultCollateral.safeTransferFrom(
+                    _tradeDataInternal._requester,
+                    address(this),
+                    _tradeDataInternal._buyInAmount
+                );
+            }
+        } else {
+            riskManager.checkAndUpdateRisks(
+                _tradeData,
+                _transformToUSD(_tradeDataInternal._buyInAmount, _tradeDataInternal._collateralPriceInUSD, transformDecimal)
```

Comment:
There are a lot of calls to transformToUSD, maybe it can be done in a batch call.

Reply:
New variables can not be added due to StackTooDeep.
The main issue is the risk assesment before ticket formation and also the events triggered

Reply:
Modified the code, added another function :
`_checkRisksAndUpdateStakingVolume`